02_Util/src/smartCharging/LocalSmartCharging.ts

```diff
@@ -0,0 +1,265 @@
+import { ISmartCharging } from './SmartCharging';
+import {
+  ChargingProfileKindEnumType,
+  ChargingProfilePurposeEnumType,
+  ChargingProfileType,
+  ChargingRateUnitEnumType,
+  ChargingSchedulePeriodType,
+  ChargingScheduleType,
+  EnergyTransferModeEnumType,
+  NotifyEVChargingNeedsRequest,
+  NotifyEVChargingScheduleRequest,
+} from '@citrineos/base';
+import {
+  IChargingProfileRepository,
+  Transaction,
+  ChargingSchedule,
+  ChargingProfile,
+} from '@citrineos/data';
+import { ILogObj, Logger } from 'tslog';
+
+export class LocalSmartCharging implements ISmartCharging {
```

Comment:
Should this be in 02_Util or rather in the smart charging module? I would think rather in the module as I don't think it would be use anywhere else?

I might be missing something, but why is this bound to local smartcharging? K03 defines local smart charging as:
> Local Smart Charging describes a use case in which smart charging enabled Charging Stations have charging limits controlled locally by a Local Controller, not directly by the CSMS.

Are we going to add an implementation for Central smart charging?

Reply:
I moved these files into the module itself.

The file name is indeed confusing. Thanks for pointing this out! It is not related to local smart charging in OCPP protocol. 
In the future, we plan to integration with external smart charging services which calculate charging profiles for us. But before integration, we create this script to achieve some basic calculation or validation. I rename it to `InternalSmartCharging`.  Please feel free to let me know if there is any better fitting name.